neuraxle/metaopt/random.py

```diff
@@ -23,22 +23,21 @@
 import copy
 from abc import ABC, abstractmethod
 from typing import List
-
+import math
 import numpy as np
 from sklearn.metrics import r2_score
 
 from neuraxle.base import MetaStepMixin, BaseStep
-from neuraxle.steps.numpy import NumpyConcatenateOuterBatch
+from neuraxle.steps.numpy import NumpyConcatenateOuterBatch, NumpyConcatenateOnCustomAxis
 from neuraxle.steps.util import StepClonerForEachDataInput
 
 
 class BaseCrossValidation(MetaStepMixin, BaseStep, ABC):
     # TODO: assert that set_step was called.
     # TODO: change default argument of scoring_function...
-    def __init__(self, scoring_function=r2_score, k_fold=3, joiner=NumpyConcatenateOuterBatch()):
+    def __init__(self, scoring_function=r2_score, joiner=NumpyConcatenateOuterBatch()):
         super().__init__()
         self.scoring_function = scoring_function
-        self.k_fold = k_fold
         self.joiner = joiner
```

Comment:
Notice that here the joiner's behavior could be changed in the next PR about the transform and inverse transform of the provided joiner. In other words, the Base Cross Validation needs an input joiner and an output joiner. 

(see the inheriting KFold class below when it joins : the joiner in this class should perhaps also do this: 
            inputs, outputs = self.joiner.transform(inputs, outputs)

Instead of this: 
            inputs = self.joiner.transform(inputs)
            outputs = self.joiner.transform(outputs)

It happens often that the output is tied to the input and needs to be processed jointly in one function call like that.

Reply:
@Eric2Hamel Also see the related issue that can change the X and y at the same time in a step: https://github.com/Neuraxio/Neuraxle/issues/30

So here if the joiner is an `OutputTransformerMixin` then it would process all at once, and otherwise it would do the first join, and then the other, as it is done right now.

Reply:
Will change the joiner's behavior in next PR.